api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -719,36 +719,6 @@ public ResponseEntity<WorkspaceUserRolesResponse> shareWorkspacePatch(
 
     WorkspaceUserRolesResponse resp = new WorkspaceUserRolesResponse();
 
-    DbUser currentUser = userProvider.get();
-    // Revoke lifescience permission before asking Firecloud to remove users; after unsharing
-    // in Firecloud, we can no longer get the user's petSA from SAM using their credentials.
-    if (dbWorkspace.getCdrVersion().getAccessTier().getEnableUserWorkflows()) {
-      List<String> finalWorkflowUsers =
-          aclsByEmail.entrySet().stream()
-              .filter(entry -> shouldGrantWorkflowRunnerAsService(currentUser, entry))
-              .map(Map.Entry::getKey)
-              .collect(Collectors.toList());
-
-      // Find the users who are owners or writers before share, but not in the new gain permission
```

Comment:
@jmthibault79 @yonghaoy just my for knowledge, why is revoking workflowrunner role need previously when sharing a workspace?

Reply:
Today, WorkflowRunner is included by Terra in the roles given to all Workspace Writers.  Before https://precisionmedicineinitiative.atlassian.net/browse/RW-7749 this role was explicitly added by RWB to users during workspace sharing.

We removed the sharing logic to add the WorkflowRunner role but we kept the **unsharing** logic to **remove** the role, because some users in some workspaces still had the explicit role added to them in addition to Workspace Writer.  When Workspace Writer was removed, they continued to have the WorkflowRunner role, which was not appropriate.

Other relevant links:
* https://precisionmedicineinitiative.atlassian.net/browse/RW-8469
* https://github.com/all-of-us/workbench/pull/6951